core/src/main/java/org/infinispan/interceptors/impl/RetryingEntryWrappingInterceptor.java

```diff
@@ -50,7 +58,12 @@ Object handleDataWriteReturn(InvocationContext ctx, DataWriteCommand dataWriteCo
    }
 
    @Override
-   protected Object setSkipRemoteGetsAndInvokeNextForManyEntriesCommand(InvocationContext ctx, WriteCommand command) {
+   protected Object setSkipRemoteGetsAndInvokeNextForManyEntriesCommand(InvocationContext ctx, WriteCommand command,
+         CompletionStage<Void> delay) {
+      if (CompletionStages.isCompletedSuccessfully(delay)) {
+         return asyncValue(delay).thenApply(ctx, command, (rCtx, rCommand, rValue) ->
```

Comment:
`delay` is already completed.

Reply:
Yeah, the if wasn't correct. It should be either when it is completed or it was completed, swapped the code branches.